spec/bolt/inventory/inventory2_spec.rb

```diff
@@ -682,6 +687,329 @@ def common_data(transport)
+              { 'name' => 'group1', 'targets' => %w[target1] }
+            ],
+            'config' => {
+              'ssh' => {
+                'user' => 'you',
+                'host-key-check' => false
+              }
+            }
+          }
+        }
+        let(:inventory) { Bolt::Inventory::Inventory2.new(data) }
+
+        it 'should return group config for an alias' do
+          target = inventory.get_target('alias1')
+          expect(target.options).to include('host-key-check' => false)
+          expect(target.user).to eq('you')
+          expect(target.config).to eq("ssh" => { "user" => "you", "host-key-check" => false })
+        end
+
+        it 'should merge config from targets' do
+          target = inventory.get_target('alias3')
+          expect(target.options).to include('host-key-check' => false)
+          expect(target.config).to eq("ssh" => { "user" => "me", "host-key-check" => false })
+        end
+
+        it 'should resolve target labels as long as a single target is returned' do
+          target = inventory.get_target('group1')
+          expect(target.name).to eq('target1')
+          expect(target.user).to eq('you')
+          expect(target.config).to eq("ssh" => { "user" => "you", "host-key-check" => false })
+        end
+      end
+
+      context 'with all options in the config' do
+        def common_data(transport)
+          {
+            'user' => 'me' + transport,
+            'password' => 'you' + transport,
+            'port' => '12345' + transport,
+            'private-key' => 'anything',
+            'ssl' => false,
+            'ssl-verify' => false,
+            'host-key-check' => false,
+            'connect-timeout' => transport.size,
+            'tmpdir' => '/' + transport,
+            'run-as' => 'root',
+            'tty' => true,
+            'sudo-password' => 'nothing',
+            'extensions' => '.py',
+            'service-url' => 'https://master',
+            'cacert' => transport + '.pem',
+            'token-file' => 'token',
+            'task-environment' => 'prod'
+          }
+        end
+
+        let(:data) {
+          {
+            'targets' => ['ssh://target', 'winrm://target', 'pcp://target', 'target'],
+            'config' => {
+              'transport' => 'winrm',
+              'modulepath' => 'nonsense',
+              'ssh' => common_data('ssh'),
+              'winrm' => common_data('winrm'),
+              'pcp' => common_data('pcp')
+            }
+          }
+        }
+        let(:conf) { Bolt::Config.default }
+        let(:inventory) { Bolt::Inventory::Inventory2.new(data, conf) }
+
+        it 'should not modify existing config' do
+          inventory.get_target('ssh://target')
+          expect(conf.transport).to eq('ssh')
+          expect(conf.transports[:ssh]['load-config']).to be true
+          expect(conf.transports[:winrm]['ssl']).to be true
+          expect(conf.transports[:winrm]['ssl-verify']).to be true
```

Comment:
One check for winrm seems like enough

Reply:
This is replicated from above. Seems harmless to explicitly ensure things are not overridden.